### DISCUSSION TITLE
SkinnedMeshRenderer 속성이 이상 합니다

### DISCUSSION THREAD

**Discussion by smilejsu82** [naverz/zepeto-studio-kor#554] [5. Q&A]
![image](https://user-images.githubusercontent.com/50188795/160624689-32cc2dc6-c863-4fd8-9214-96a4263a4d28.png)

![image](https://user-images.githubusercontent.com/50188795/160624675-5f228bb4-4099-4395-9819-8bd9323990a6.png)

![image](https://user-images.githubusercontent.com/50188795/160624557-dc41f20f-702b-4ced-977a-8556df6d3770.png)

메터리얼들을 출력해보려고 했는데 undefined 나오네요 
무엇을 잘못 했을까요?
왜 이런 현상이 발생하는것인가요?

**Comment by minjujuu** [comment]
material.name 으로 출력해보세요!
```
console.log(`${ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.GetComponentInChildren<SkinnedMeshRenderer>().material.name}`);
```
![image](https://user-images.githubusercontent.com/79794414/160730394-087b0bcf-42f4-4e4d-8233-0f123122782e.png)

**Reply by smilejsu82** [reply]
Material 이 여러개면 어떻게 해야 하나요?
![image](https://user-images.githubusercontent.com/50188795/160730587-cba43a84-240f-44ec-8e7a-993fdbf6c298.png)

**Reply by minjujuu** [reply]
작성해주신 코드에서 renderer.materialsl[i].name 으로 하면 여러 개 출력될 것으로 보입니다!
예시로 캐릭터의 material들을 출력하기 위해 다음과 같이 코드를 작성했습니다.
```
ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.GetComponentsInChildren<SkinnedMeshRenderer>().forEach(element => {
        console.log(element.material.name);
});
```
![image](https://user-images.githubusercontent.com/79794414/160730892-6b1416ad-46a9-4d84-9e28-8bd4177682cb.png)

**Reply by smilejsu82** [reply]
답변 감사합니다 
작성하신 GetComponentsInChildren<SkinnedMeshRenderer>() 이부분은 SkinnedMeshRenderer컴포넌트가 자식으로 여러개 인가요?

전 SkinnedMeshRenderer가 하나인데.. 
하나의 SkinnedMeshRenderer안에 여러개의 Material들이 있는 형태 입니다 
이것은 동작 하나요?

![image](https://user-images.githubusercontent.com/50188795/160731360-32e51bea-a58b-40ee-b27c-8dab3cf5784f.png)


forEach를 사용하기 위해 선행 해야 할것이 있나요?
![image](https://user-images.githubusercontent.com/50188795/160731450-4504d135-b061-4808-9e36-376dacacc7ea.png)

![image](https://user-images.githubusercontent.com/50188795/160731463-05c6e384-ba17-47cf-82e0-6b6e76522d42.png)

**Reply by smilejsu82** [reply]
또한 materials 배열에 2개가 있는것을 확인했는데 
for문을 돌리면 왜 undefined가 나오는것일까요?
![image](https://user-images.githubusercontent.com/50188795/160731716-636b0bd8-012b-42dc-8202-44580c53c9bb.png)


혹시 for문과 배열을 잘못 사용하고 있나요?
![image](https://user-images.githubusercontent.com/50188795/160731764-42d59d75-0eb6-42fb-bff8-527eddd5fc5b.png)

**Reply by minjujuu** [reply]
제가 다르게 이해를 했었는데, 아래와 같이 한 번 시도해보시면 좋을 것 같습니다!
```
        for(let i =0;i<this.renderer.materials.length;i++)
        {
            console.log(`mat name : ${this.renderer.materials.get_Item(i).name}`);
        }
```

**Reply by smilejsu82** [reply]
와우 ~ 잘 동작 합니다  👍 
![image](https://user-images.githubusercontent.com/50188795/160735924-0a0f7c69-afd9-4579-a56c-f710e15ab55f.png)





배열의 요소를 인덱스로 접근 할수 없나요?
![image](https://user-images.githubusercontent.com/50188795/160742861-91fe84b1-68f2-4e39-a4b7-7192196211f6.png)
![image](https://user-images.githubusercontent.com/50188795/160742875-9fd4dbe4-ef7d-4e8c-b5d8-812a189d58c6.png)

**Reply by minjujuu** [reply]
다른 경우에는 보여주신 코드처럼 배열의 요소를 [i]로 가져오는 것이 가능한데,
간혹 SkinnedMeshRenderer의 materials를 처럼 [i]로 가져올 수 없는 경우 get_Item(i)으로 사용하여 개발할 수 있을 것 같습니다!

**Reply by smilejsu82** [reply]
답변 감사합니다 ~ 해결 완료 ! :)
